feat: UIG-2749 - vl-header - logout request

ksvlaanderen opened the pull request:
Afhandelen van logout request toegevoegd
Storybook uitgebreid
Cypress testen herschreven

Ik heb 2 manieren voorzien om het logout request af te handelen:
- **data-vl-reject-logout attribuut**: dit zorgt ervoor dat alle logouts (behalve die door de gebruiker) afgewezen worden. Dit is voor Freek het gevraagde gedrag en dat kan hij nu heel simpel door een attribuut te plaatsen
- **logoutCallback property**: hiermee kan je het heft in eigen handen nemen en obv de logout reason zelf bepalen of je wilt uitloggen of niet. Dit is een asynchrone methode die een boolean terug verwacht zodat de afnemer een call naar hun backend kan doen om bv. te kijken of hun sessie nog geldig is. Ook hier worden logout requests door de gebruiker nooit afgewezen.

[Jira](https://www.milieuinfo.be/jira/browse/UIG-2749)
[Bamboo](https://www.milieuinfo.be/bamboo/browse/UIGOV-CUWC353)

Goldflow commented on libs/sections/src/header/stories/vl-header.stories-doc.mdx:
```diff
@@ -26,18 +26,34 @@ import { VlHeader } from '@domg-wc/sections';
+Een logout request door de gebruiker wordt nooit afgewezen.
+
 
+Zie ook: [De aanvragen voor applicatie logout afhandelen via JavaScript](https://vlaamseoverheid.atlassian.net/wiki/spaces/IKPubliek/pages/6336119448/Aanmelden+met+eenvoudig+of+gekoppeld+toegangsbeheer#De-aanvragen-voor-applicatie-logout-afhandelen-via-JavaScript)
```
puike documentatie! 💯

ksvlaanderen replied:
Learned from the best! ❤️

Goldflow commented on libs/sections/src/header/vl-header.section.ts:
```diff
@@ -19,35 +19,55 @@ const customRegistration = () =>
             customElements.define('vl-header', VlHeader);
         });
 
+export const headerDefaults = {
```
👍🏼

ksvlaanderen committed:
feat: UIG-2749 - vl-header - logout request

Afhandelen van logout request toegevoegd
Storybook uitgebreid
Cypress testen herschreven

kspeltix commented:
:tada: This PR is included in version 1.33.0 :tada:

The release is available on [GitHub release](https://github.com/milieuinfo/uigov-web-components/releases/tag/v1.33.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

kspeltix commented:
:tada: This PR is included in version 1.33.0 :tada:

The release is available on [GitHub release](https://github.com/milieuinfo/uigov-web-components/releases/tag/v1.33.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
